apollo-runtime/src/main/java/com/apollographql/android/cache/normalized/GraphQLResponseNormalizer.java

```diff
@@ -0,0 +1,137 @@
+package com.apollographql.android.cache.normalized;
+
+import com.apollographql.android.api.graphql.Field;
+import com.apollographql.android.api.graphql.Operation;
+import com.apollographql.android.cache.ResponseReaderShadow;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+import javax.annotation.Nullable;
+
+public class GraphQLResponseNormalizer implements ResponseReaderShadow {
+
+  private Stack<List<String>> pathStack;
+  private Stack<ApolloRecord> recordStack;
+  private Stack<Object> valueStack;
+  private Set<String> dependentKeys;
+  private final CacheKeyResolver cacheKeyResolver;
+
+  private List<String> path;
+  private ApolloRecord currentRecord;
+
+  private InMemoryNormalizedCacheStore recordSet;
```

Comment:
I wasn't sure if I wanted to add `allRecords` to the generic interface, which is necessary for the `NormalizedCacheStore`. This is really more of a per-request store that will be thrown out once parsing is done (and the results added to the aggregate store the user supplies)